engine/src/main/java/org/terasology/persistence/typeHandling/extensionTypes/ModifiableValueTypeHandler.java

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.persistence.typeHandling.extensionTypes;
+
+import gnu.trove.list.TFloatList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.persistence.typeHandling.PersistedData;
+import org.terasology.persistence.typeHandling.PersistedDataArray;
+import org.terasology.persistence.typeHandling.PersistedDataSerializer;
+import org.terasology.persistence.typeHandling.RegisterTypeHandler;
+import org.terasology.utilities.modifiable.ModifiableValue;
+
+import java.util.Optional;
+
+@RegisterTypeHandler
+public class ModifiableValueTypeHandler extends org.terasology.persistence.typeHandling.TypeHandler<ModifiableValue> {
```

Comment:
I'd like to have some JavaDoc here (or some documentation somewhere else) on how the `ModifiableValue` is actually persisted. What is the format to specify a modifiable in a prefab?

From looking at the code it seems like the value is persisted as "unlabeled" array:

```
"<field_name>": [<base_value>, <pre_modifier>, <multiplier>, <post_modifier>]
```

I don't think this is very intuitive, and when seeing this in a prefab I'd think of a vector or list but not of a special data class 🙈

Reply:
Added Javadocs, the default way to specify in prefabs is with a single value only and then the modifiers are given default values.